codalab/lib/metadata_util.py

```diff
@@ -78,12 +77,16 @@ def request_missing_metadata(bundle_subclass, args, initial_metadata=None):
         if not new_initial_metadata[spec.key]:
             default = MetadataDefaults.get_default(spec, bundle_subclass, args)
             new_initial_metadata[spec.key] = default
-    initial_metadata = new_initial_metadata
+    return new_initial_metadata
 
-    # If args.edit exists (when doing 'cl edit'), then we want to show
-    # the editor.
+
+def request_missing_metadata(bundle_subclass, args, initial_metadata=None):
+    '''
+    For any metadata arguments that were not supplied through the command line,
+    pop up an editor and request that data from the user.
+    '''
     if not getattr(args, 'edit', True):
-        return initial_metadata
+        return fill_missing_metadata(bundle_subclass, args, initial_metadata)
```

Comment:
Oh yeah. This is what I was thinking:

When there are no args supplied with edit eg. ``cl edit b.txt``, the condition ``not getattr(args, 'edit', True)`` will evaluate to False and the editor will be shown, which is the intended behavior.

When args are supplied with edit eg. `` cl edit b.txt -n c.txt ``, request_missing_metadata method would never be called (since ``is_new_metadata_updated`` in bundles_cli: do_cl_edit will be True)

When args are supplied with edit but with the same name e.g. `` cl wedit c.txt -n c.txt``, again 
`` is_new_metadata_updated`` will evaluate to True and will prevent request_missing_metadata method from being called. 

However, let's look at some other command. Let's say when we call ``cl run 'echo hello'``, the condition ``not getattr(args, 'edit', True)`` will evaluate to True and the editor will not be shown, which is the intended behavior. :)